python/tank/bootstrap/threading.py

```diff
@@ -0,0 +1,295 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+try:
+    # Import Qt without having to worry about the version to use.
+    from ..util.qt_importer import QtImporter
+    importer = QtImporter()
+    QtCore = importer.QtCore
+    QtGui = importer.QtGui
+except ImportError:
+    # Raise an exception when Qt is not available.
+    raise
```

Comment:
i might be missing something, but what's the purpose of this? You are trapping an exception just to re-raise it?